config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -313,5 +403,50 @@ private Saml2AuthenticationRequestFactory getResolver(B http) {
 		}
 	}
 
+	/**
+	 * Configuration held for the default authentication manager used in SAML authentication.
+	 */
+	public class AuthenticationConfigurationSpec<B extends HttpSecurityBuilder<B>> {
```

Comment:
I'd recommend leaving this out for the time being. The main reason is that applications can either call  `authenticationManager()` or they can use a post-processor.

A second reason is that the DSL is not typically a place where all the properties of a given component are exposed.

Reply:
The [`Oauth2LoginConfigurer`](https://github.com/spring-projects/spring-security/blob/65f6025cef07113c4a165dcc9ad6c50cd1c66475/config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurer.java#L512-L515) chooses to use beans or shared objects to configure the same.
Would that be preferred if one doesn't want to override the default implementation

Reply:
Should I expose them as `shared objects` (like `Oauth2LoginConfigurer` does) or do we simply close the original issue as `won't fix` since the postprocessor does allow you to set it?

Reply:
I've opted to simply document the post processor approach in the documentation.